### PR TITLE
Add support for local priority in presets

### PR DESCRIPTION
**Pull request by lukasIO**
In order to satisfy TS I had to update to v5.0.0 which makes this PR a bit unwieldy. 
TS 5.0 changes are only to make type imports explicit (we had this setting before already, but 5.0 caught a lot more cases)

This PR adds support to construct video and audio presets with an additional `priority` option which (on supported browsers) tells the browser about how to distribute available resources (cpu + network) between different tracks.

- deprecates `audioBitrate` setting on `TrackPublishDefaults` in favor of `audioPreset` that now includes `priority`
- by default audio tracks will get the priority `high`, screen share video tracks `medium` and camera video tracks `low`

### CODE REVIEW

**Review by davidzhao — APPROVED** [review]
lg, how well does the priorities work in practice? does it show a visible improvement when using NLC?

File: src/room/participant/LocalParticipant.ts [hunk 1]
```diff
@@ -651,10 +654,11 @@ export default class LocalParticipant extends Participant {
         opts,
       );
       req.layers = videoLayersFromEncodings(req.width, req.height, simEncodings ?? encodings);
-    } else if (track.kind === Track.Kind.Audio && opts.audioBitrate) {
+    } else if (track.kind === Track.Kind.Audio) {
       encodings = [
         {
-          maxBitrate: opts.audioBitrate,
+          maxBitrate: opts.audioPreset?.maxBitrate ?? opts.audioBitrate,
```
[the remark this answers is not in the document]
Comment (lukasIO) [reply src/room/participant/LocalParticipant.ts:660]: opts gets merged with trackPublishDefaults in `publishTrack` so it should always be defined!

File: src/room/PCTransport.ts [hunk 2]
```diff
@@ -35,9 +35,12 @@ export default class PCTransport extends EventEmitter {
 
   onOffer?: (offer: RTCSessionDescriptionInit) => void;
 
-  constructor(config?: RTCConfiguration) {
+  constructor(config?: RTCConfiguration, mediaConstraints: Record<string, unknown> = {}) {
     super();
-    this.pc = new RTCPeerConnection(config);
+    this.pc = isChromiumBased()
```
[the remark this answers is not in the document]
Comment (lukasIO) [reply src/room/PCTransport.ts:40]: ah, of course, thanks!

**Comment by lukasIO** [comment]
> how well does the priorities work in practice? does it show a visible improvement when using NLC?

just did some tests on chrome: 
the target bitrate of lower priority layers changes more significantly when others with higher priority are published as well while, in comparison, layers with higher priority tend to have less fluctuations in the target bitrate.

**Pull request merged**